RecipeMatch/Views/DraggableViewBackground.m

```diff
@@ -136,7 +142,12 @@ - (void)swipeRight{
     [UIView animateWithDuration:0.2 animations:^{
         dragView.overlayView.alpha = 1;
     }];
-    [dragView rightClickAction];
+    [dragView rightClickAction:^(BOOL succeeded, NSError *error){  // check index after card is dismissed from view
+        if(succeeded){
+            self->allCardsIndex++;
+            [self checkCardIndexStatus];
+        }
+    }];
```

Comment:
i noticed that you call this `rightClickAction ` from `DraggableViewBackground`, then inside `rightClickAction` (that is in `DraggableView`) you call `delegate draggableViewCardSwipedRight`, which is in.... `DraggableViewBackground`..... do you see where i'm going with it?